docs/src/chapters/testy.adoc

```diff
@@ -1,12 +1,278 @@
 [#_testy]
 == Testy
+Aby już na wczesnym etapie wykrywać powstałe błędy, zespół zdecydował o tworzeniu testów podczas pisania poszczególnych
+elementów systemu. Dzięki takiemu podejściu udało się uniknąć wielu błędów w trakcie rozbudowy aplikacji. Ułatwiło to
+również dokonywanie _code review_, ponieważ osoba sprawdzająca mogła w łatwy sposób zweryfikować czy dana funkcjonalność
+działa w taki sposób, w jaki została zaplanowana.
+Podczas implementacji należało pamiętać również o tym, że tworzony kod musi być testowalny, w konsekwencji czego był on
+czytelniejszy i prostszy do zrozumienia.
 
-=== Serwer
+W celu zautomatyzowania procesu uruchamiania testów, członkowie zespołu zdecydowali się na użycie narzędzia
+_ciągłej integracji_ Github Actions.
+Wystarczyło tylko stworzyć odpowiedni skrypt _YAML_, który czuwał nad tym, aby każdorazowo uruchomić testy po
+wypchnięciu zmian do repozytorium.
+Katalog _workflows_ z plikami prezentuje się następująco:
 
-=== Aplikacja webowa
+.Zawartość folderu workflows z plikami YAML
+image::../images/tests_chapter/workflows_CI.png[]
 
-"ciężko stwierdzić granicę pomiędzy testami jednostkowymi a integracyjnymi" - autorzy react testing library
+Na poniższym zrzucie ekranu można zauważyć listę kroków wykonywanych każdorazowo po wypchnięciu zmian do części
+backendowej aplikacji. Testy w tym wypadku uruchamiane są przez narzędzie _Maven_.
 
-=== Aplikacja mobilna
+.Fragment pliku webservice.yml prezentujący listę wykonywanych kroków
+[source, yaml]
+----
+include::../../../.github/workflows/webservice.yml[lines=17..37]
+----
+
+=== Testy jednostkowe i integracyjne
+
+==== Serwer
+Kod źródłowy dla testów jednostkowych i integracyjnych umieszczony został w katalogu _test_, znajdującym się w folderze
+_src_. Testy, tak samo jak główny kod _Javy_, podzielone zostały na pakiety pogrupowane według funkcjonalności. Ich
+układ można zaobserwować na zrzucie ekranu poniżej.
+
+.Pakiety testów Java dla serwera
+image::../images/tests_chapter/test_folders_server.png[]
+
+W każdym z widocznych katalogów utworzone zostały foldery, gdzie umiejscowiono pliki z testami.
+
+* admins- Katalog _admins_ zawiera w sobie 4 foldery oraz 5 plików z testami co jest widoczne na poniższym zrzucie
+ekranu. Jak sama nazwa wskazuje, znajdują się tam testy związane z administratorami np. sprawdzanie, czy ich tworzenie
+działa prawidłowo.
+
+.Katalog admins
+image::../images/tests_chapter/admins_folder.png[]
+
+* auth- Zawiera 7 plików z testami rozmieszczonymi w 5 folderach. Sprawdzane jest tutaj m.in. działanie
```

Comment:
rzucił mi się w oczy brak spacji przed myślnikiem, jakoś dziwnie to wygląda jak na to patrzysz

Reply:
zmieniłem